src/west/commands/command.py

```diff
@@ -100,6 +109,22 @@ def add_parser(self, parser_adder):
 
         return self.parser
 
+    def check_up_to_date(self):
+        '''Checks if west itself is up to date, emitting a warning if not.'''
+        try:
+            manifest = Manifest.from_file(sections=['west'])
+            west_project = manifest.west_project
+            if not west_project.is_up_to_date():
+                log.wrn(west_project.format(
+                    '{name_and_path} is at {wsh}, '
+                    'which is not up to date with manifest revision {msh}; '
+                    'consider a "west selfupdate"',
+                    wsh=west_project.sha('HEAD'),
+                    msh=west_project.revision))
+        except:                 # noqa: E722
```

Comment:
Part two of SHA fix.
```suggestion
        except ValueError:
            # Revision is not a SHA, ignore.
            pass
        except:                 # noqa: E722
```